lib/chef/resource/locale.rb

```diff
@@ -99,6 +104,15 @@ def generate_locales
           shell_out!("locale-gen #{unavailable_locales.join(" ")}")
         end
 
+        # Windows only: Sets the system locale for the current computer.
+        # @see https://docs.microsoft.com/en-us/powershell/module/internationalcmdlets/set-winsystemlocale
+        #
+        def set_system_locale
+          response = powershell_exec("Set-WinSystemLocale -SystemLocale #{new_resource.lang}")
+
+          raise response.errors.join(" ") if response.error?
+        end
```

Comment:
@kapilchouhan99 looks like it might not be idempotent. Considering it only running Set query and not checking existing values,

Reply:
We should have idempotency check